### PR TITLE
feat(basiccell): BasicCell Component Implementation

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**BasicCell**
Shortcut ticket: [sh-208409](https://app.shortcut.com/sci-design-system/story/208409/basiccell-component)

Implemented with &lt;td&gt; under the hood.

## Checklist

- [ ] Default Story in Storybook
- [ ] LivePreview Story in Storybook
- [ ] Test Story in Storybook
- [ ] Tests written
- [ ] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

**Review by clarsen-czi — APPROVED** [review]
Looks awesome, @masoudmanson !!

File: src/core/BasicCell/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import { Args, Story } from "@storybook/react";
+import * as React from "react";
+import InputText from "../InputText";
+import BasicCellRaw from "./index";
+
+const BasicCell = (props: Args): JSX.Element => {
+  const [maxWidth, setMaxWidth] = React.useState(100);
+
+  const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMaxWidth(+event.target.value);
```
Comment (mmmmmmaya) [inline src/core/BasicCell/index.stories.tsx:11]: I haven't seen the notation with `+` before. what does that indicate?

File: src/core/BasicCell/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  };
+
+  return (
+    <table>
+      <tbody>
+        <tr>
+          {/* <span>Set a max-width to see text wrap inside the cell</span> */}
```
Comment (mmmmmmaya) [inline src/core/BasicCell/index.stories.tsx:18]: do we need this if it's commented out?

File: src/core/BasicCell/style.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    return `
+        font-family: ${typography?.fontFamily};
+        padding: ${spacings?.l}px ${spacings?.s}px;
+        text-align: ${textPosition};
+        min-width: 96px;
+        max-width: 100%;
+        display: block;
+        overflow: hidden;
+        border: dashed 1px #ddd;
```
Comment (mmmmmmaya) [inline src/core/BasicCell/style.ts:46]: do we have a predefined border for this in getBorders?

**Review by mmmmmmaya — APPROVED** [review]

File: src/core/BasicCell/index.stories.tsx (hunk 1 above)
Comment (masoudmanson) [reply src/core/BasicCell/index.stories.tsx:11]: It is the same as `parseInt()`. The `event.target.value` is a type of string and `+` converts it to a number. 🥸

File: src/core/BasicCell/index.stories.tsx (hunk 2 above)
Comment (masoudmanson) [reply src/core/BasicCell/index.stories.tsx:18]: Eew, no we don't :D I'll remove it.

File: src/core/BasicCell/style.ts (hunk 3 above)
Comment (masoudmanson) [reply src/core/BasicCell/style.ts:46]: This was for preview purposes. I forgot to remove it. Nice catch 👀

**Pull request merged**